app/src/main/java/com/duckduckgo/app/browser/navigation/bar/BrowserNavigationBarViewIntegration.kt

```diff
@@ -109,4 +110,19 @@ class BrowserNavigationBarViewIntegration(
     private fun onDisabled() {
         keyboardVisibilityJob?.cancel()
     }
+
+    fun setContentCanScroll(
+        canScrollUp: Boolean,
+        canScrollDown: Boolean,
+        topOfPage: Boolean,
+    ) {
+        val shouldShowShadows = (canScrollUp || canScrollDown) && !topOfPage
+        if (omnibar.omnibarPosition == OmnibarPosition.TOP) {
+            if (shouldShowShadows) {
+                navigationBarView.showShadow(true)
+            } else {
+                navigationBarView.showShadow(false)
+            }
+        }
```

Comment:
```suggestion
        if (omnibar.omnibarPosition == OmnibarPosition.TOP) {
            navigationBarView.showShadow(canScrollDown)
        }
```

@malmstein shouldn't the bottom nav bar just check if it can scroll down to show the shadow?

Reply:
good idea!